.github/workflows/build-lint-test.yml

```diff
@@ -101,14 +101,119 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: 20
-      - name: Install npm dependencies
+
+      - name: Cache node_modules
+        uses: actions/cache@v4
+        id: cache-node-modules
+        with:
+          path: |
+            frontend/node_modules
+          key: modules-${{ hashFiles('frontend/package-lock.json') }}
+
+      - name: Cache Playwright binaries
+        uses: actions/cache@v4
+        id: cache-playwright
+        with:
+          path: |
+            ~/.cache/ms-playwright
+          key: playwright-${{ hashFiles('frontend/package-lock.json') }}
```

Comment:
Did you check if this improves performance? The cache also needs to be downloaded and unpacked, so the performance improvement might be negligible or even non-existent.

Reply:
It results in a significant improvement. From my fork:
![image](https://github.com/user-attachments/assets/8aa44cb2-3dac-463c-bc7e-cbe497588647)

10s to download and unpack the cache. 30s/37s to run the UI tests on Firefox/Chrome. 1m10s to install the Webkit dependencies and run the tests on it.

(I intend to put all the UI tests back in a single job, because that probably still takes less time than the d2d tests job on Webkit.)

Reply:
Because caching does not work for WebKit, we need to install it on each runner. And doing that is barely faster than installing all three browser engines. And using cached node modules vs running `npm ci` also takes about the same time.

So until WebKit chaching works, there's no value in caching node modules and browser engines for Playwright tests. (Or until we care about a 10-ish seconds in difference.)

Reply:
The Playwright `app` tests also magically start passing again if you don't use cache. :man_shrugging: